dipy/reconst/dki.py

```diff
@@ -19,13 +19,353 @@
+    An = A0.copy()
+    Q = (errtol/4.) ** (-1/6.) * np.max(np.abs([An - xn, An - yn, An - zn]),
+                                        axis=0)
+    sum_term = np.zeros(x.shape, dtype=x.dtype)
+    n = np.zeros(x.shape)
+
+    # Convergence has to be done voxel by voxel
+    index = ndindex(x.shape)
+    for v in index:
+        # Convergence condition
+        while 4.**(-n[v]) * Q[v] > abs(An[v]):
+            xnroot = np.sqrt(xn[v])
+            ynroot = np.sqrt(yn[v])
+            znroot = np.sqrt(zn[v])
+            lamda = xnroot*(ynroot + znroot) + ynroot*znroot
+            sum_term[v] = sum_term[v] + 4.**(-n[v]) / (znroot * (zn[v]+lamda))
+            n[v] = n[v] + 1
+            xn[v] = (xn[v]+lamda) * 0.250
+            yn[v] = (yn[v]+lamda) * 0.250
+            zn[v] = (zn[v]+lamda) * 0.250
+            An[v] = (An[v]+lamda) * 0.250
+
+    # post convergence calculation
+    X = (A0 - x) / (4.**(n) * An)
+    Y = (A0 - y) / (4.**(n) * An)
+    Z = - (X+Y) / 3.
+    E2 = X*Y - 6.*Z*Z
+    E3 = (3.*X*Y - 8.*Z*Z) * Z
+    E4 = 3. * (X*Y - Z*Z) * Z**2.
+    E5 = X * Y * Z**3.
+    RD = \
+        4**(-n) * An**(-3/2.) * \
+        (1 - 3/14.*E2 + 1/6.*E3 + 9/88.*(E2**2) - 3/22.*E4 - 9/52.*E2*E3 +
+         3/26.*E5) + 3*sum_term
+
+    return RD
+
+
+def _F1m(a, b, c):
+    """ Helper function that computes function $F_1$ which is required to
+    compute the analytical solution of the Mean kurtosis.
+
+    Parameters
+    ----------
+    a : ndarray
+        Array containing the values of parameter $\lambda_1$ of function $F_1$
+    b : ndarray
+        Array containing the values of parameter $\lambda_2$ of function $F_1$
+    c : ndarray
+        Array containing the values of parameter $\lambda_3$ of function $F_1$
+
+    Returns
+    -------
+    F1 : ndarray
+       Value of the function $F_1$ for all elements of the arrays a, b, and c
+
+    Notes
+    --------
+    Function $F_1$ is defined as [1]_:
+
+    .. math::
+
+        F_1(\lambda_1,\lambda_2,\lambda_3)=
+        \frac{(\lambda_1+\lambda_2+\lambda_3)^2}
+        {18(\lambda_1-\lambda_2)(\lambda_1-\lambda_3)}
+        [\frac{\sqrt{\lambda_2\lambda_3}}{\lambda_1}
+        R_F(\frac{\lambda_1}{\lambda_2},\frac{\lambda_1}{\lambda_3},1)+\\
+        \frac{3\lambda_1^2-\lambda_1\lambda_2-\lambda_2\lambda_3-
+        \lambda_1\lambda_3}
+        {3\lambda_1 \sqrt{\lambda_2 \lambda_3}}
+        R_D(\frac{\lambda_1}{\lambda_2},\frac{\lambda_1}{\lambda_3},1)-1 ]
+
+    References
+    ----------
+    .. [1] Tabesh, A., Jensen, J.H., Ardekani, B.A., Helpern, J.A., 2011.
+           Estimation of tensors and tensor-derived measures in diffusional
+           kurtosis imaging. Magn Reson Med. 65(3), 823-836
+    """
+    # Eigenvalues are considered equal if they are not 2.5% different to each
+    # other. This value is ajusted according to the analysis reported in:
+    # http://gsoc2015dipydki.blogspot.co.uk/2015/08/rnh-post-13-start-wrapping-up-test.html
+    er = 2.5e-2
```

Comment:
It is important to keep this at 2.5% for a correct identification of the mean kurtosis analytical solution singularities ([see here](http://gsoc2015dipydki.blogspot.co.uk/2015/08/rnh-post-13-start-wrapping-up-test.html)). Therefore, I think we should keep this hardcoded to avoid users setting this parameter in a wrong way.